tests/core/test_error_handling.py

```diff
@@ -0,0 +1,191 @@
+# Standard Library
+import logging
+import os
+from pathlib import Path
+
+# Third Party
+import pytest
+
+# First Party
+from smdebug.core.error_handler import BASE_ERROR_MESSAGE
+from smdebug.core.logger import DuplicateLogFilter, get_logger
+from smdebug.core.utils import error_handler
+
+
+@pytest.fixture
+def stack_trace_filepath(out_dir):
+    return f"{out_dir}/tmp.log"
+
+
+@pytest.fixture
+def runtime_error_message():
+    return (
+        "If this RuntimeError causes the test to fail, the error handler failed to catch the error!"
+    )
+
+
+@pytest.fixture
+def value_error_message():
+    return (
+        "If this ValueError causes the test to fail, the error handler failed to catch the error!"
+    )
+
+
+@pytest.fixture
+def dummy_clean_function():
+    @error_handler.catch_smdebug_errors()
+    def do_nothing():
+        pass
+
+    return do_nothing
+
+
+@pytest.fixture
+def dummy_error_function(runtime_error_message):
+    @error_handler.catch_smdebug_errors()
+    def raise_error():
+        raise RuntimeError(runtime_error_message)
+
+    return raise_error
+
+
+@pytest.fixture
+def dummy_clean_function_with_return_val():
+    @error_handler.catch_smdebug_errors(default_return_val=False)
+    def do_nothing():
+        return True
+
+    return do_nothing
+
+
+@pytest.fixture
+def dummy_error_function_with_return_val(value_error_message):
+    @error_handler.catch_smdebug_errors(default_return_val=False)
+    def raise_error():
+        raise ValueError(value_error_message)
+
+    return raise_error
+
+
+@pytest.fixture(autouse=True)
+def set_up_logging_and_error_handler(out_dir, stack_trace_filepath):
+    """
+    Setup up each test to:
+        - Add a logging handler to write all logs to a file (which will be used to verify caught errors in the tests)
+        - Remove the duplicate logging filter
+        - Reset the error handler after the test so it that it is reenabled.
```

Comment:
Should be "Reset the error handler after the test so it is reenabled".

Reply:
Done.